src/NuGet.Core/NuGet.ProjectModel/LockFileDependencyProvider.cs

```diff
@@ -38,10 +38,9 @@ public LockFileDependencyProvider(LockFile lockFile)
             }
         }
 
-        public bool SupportsType(string libraryType)
+        public bool SupportsType(LibraryTypeFlag libraryType)
         {
-            return string.IsNullOrEmpty(libraryType) ||
-                   string.Equals(libraryType, LibraryTypes.Package);
+            return (libraryType & LibraryTypeFlag.Package) == LibraryTypeFlag.Package;
```

Comment:
Can we use `libraryType.HasFlag(LibraryTypeFlag.Package)` here? It's a little more readable IMO.